lib/capybara/queries/text_query.rb

```diff
@@ -31,6 +32,16 @@ def failure_message
           message << " but found #{@count} #{Capybara::Helpers.declension('time', 'times', @count)}"
         end
         message << " in #{@actual_text.inspect}"
+
+        if @node and visible?
```

Comment:
I don't think this message doesn't really make sense to appear in the negative failure message error